### DISCUSSION TITLE
Help interpreting residual correlations of gamma model

### DISCUSSION THREAD

**Discussion by smithja16** [JenniNiku/gllvm#130] [Q&A]
Hello,

I'm using gllvm to do a hurdle model, and the probit part works fine, but I don't understand the results for the gamma (positive only) component. The residual correlations tend to very high - often most are +1 or -1. 

Below is some test code to repeat this. Given that each 'species' is random, the residual correlations should be very small. If the same code is run with 'family="gaussian" the residual correlations are very low, as expected.

set.seed(321)
YData <- matrix(rgamma(n=50*6, shape=1.5, rate=3), nrow=50, ncol=6)
XData <- data.frame(Depth=rnorm(n=50, mean=100, sd=20),
                    Location=as.factor(c(rep("North",25), rep("South",25))))

Mx <- gllvm(y=YData, X=XData,
                  formula= ~ Depth + Location,
                  family="gamma",
                  num.lv=1)
crx <- getResidualCor(Mx)
corrplot(crx, diag=F, type="lower", method="square")

![Rplot](https://github.com/JenniNiku/gllvm/assets/6924440/a022190e-2524-45ae-a402-a0b39e52d0dc)

Thanks for any help!
James

**Comment by BertvanderVeen** [comment]
Hello James.

What you are seeing is most likely due to low values for the species-specific variances in the residual covariance matrix. This can be completely normal, due to the properties of the dataset you are looking at, or perhaps because the number of latent variables that you have chosen does not provide for a sufficiently flexible model, or in contrast, perhaps it is too flexible and the LVs are not important in explaining the associations between your species. 

One thing to also always consider is if your model has converged properly; make sure to always refit with multiple sets of starting values as with the `n.init` argument.
